src/pages/SignUp/SignUp.js

```diff
@@ -1,7 +1,148 @@
 import React, { Component } from 'react';
+import { chkEmail, chkPwd, chkName, chkAddress } from './Validation';
+import '../SignUp/SignUp.scss';
 
 export default class Login extends Component {
+  emailRef = React.createRef();
+  pwRef = React.createRef();
+
+  state = {
+    name: '',
+    email: '',
+    pw: '',
+    address: '',
+  };
+
+  emailInput = email => {
+    this.setState({
+      email: email.target.value,
+    });
+  };
+
+  pwInput = pw => {
+    this.setState({
+      pw: pw.target.value,
+    });
+  };
+
+  nameInput = name => {
+    this.setState({
+      name: name.target.value,
+    });
+  };
+
+  addressInput = address => {
+    this.setState({
+      address: address.target.value,
+    });
+  };
+
+  checkAll = e => {
+    if (chkName(this.state.name) === false) {
+      alert('이름을 입력해주세요.');
+    }
+
+    if (chkEmail(this.state.email) === false) {
+      alert('이메일 형식이 유효하지 않습니다.');
+      this.emailRef.current.value = '';
+    }
+
+    if (chkPwd(this.state.pw) === false) {
+      alert('영문,숫자를 혼합하여 6~12자 이내로 만들어주세요.');
+      this.pwRef.current.value = '';
+    }
+
+    if (chkAddress(this.state.address) === false) {
+      alert('주소를 입력해주세요.');
+    }
+
+    if (
+      chkName(this.state.name) &&
+      chkEmail(this.state.email) &&
+      chkPwd(this.state.pw) &&
+      chkAddress(this.state.address) === true
+    ) {
+      alert('회원가입 완료');
+    } else {
+      alert('입력란을 다시 확인하세요.');
+    }
+  };
+
+  handleKeyPress = e => {
+    if (e.key === 'Enter') {
+      this.checkAll();
+    }
+  };
+
   render() {
-    return <div></div>;
+    return (
+      <div className="wholePage">
+        <header className="Nav">임시헤더</header>
+        <div className="signUpContents">
+          <h1 className="signUpTitle">CREATE AN ACCOUNT</h1>
+          <span className="indicater">* indicates a required field</span>
+          <div className="signUpInput">
+            <label className="info">
+              FULL NAME
+              <span className="dot">*</span>
+            </label>
+            <input
+              className="userInput"
+              type="text"
+              onChange={this.nameInput}
+              onKeyPress={this.handleKeyPress}
+            ></input>
+          </div>
+          <div className="signUpInput">
+            <label className="info">
+              EMAIL
+              <span className="dot">*</span>
+            </label>
+            <input
+              className="userInput"
+              type="text"
+              ref={this.emailRef}
```

Comment:
```suggestion
            <input
              className="userInput"
              type="text"
              value={email}
```
- ref를 사용하지 않고 state에 저장된 value를 사용하면 어떨까요?

Reply:
> * ref를 사용하지 않고 state에 저장된 value를 사용하면 어떨까요?

ref 제거 후 value 사용으로 수정하였습니다.